nuget/lib/dependabot/nuget/nuget_client.rb

```diff
@@ -49,38 +58,32 @@ def self.get_package_versions(dependency_name, repository_details)
         matching_versions
       end
 
+      sig { params(repository_details: T::Hash[Symbol, String]).returns(T.nilable(T::Array[String])) }
       private_class_method def self.get_versions_from_versions_url_v3(repository_details)
-        body = execute_json_nuget_request(repository_details[:versions_url], repository_details)
+        body = execute_json_nuget_request(repository_details.fetch(:versions_url), repository_details)
         body&.fetch("versions")
       end
 
+      sig { params(repository_details: T::Hash[Symbol, String]).returns(T.nilable(T::Set[String])) }
       private_class_method def self.get_versions_from_registration_v3(repository_details)
-        url = repository_details[:registration_url]
+        url = repository_details.fetch(:registration_url)
         body = execute_json_nuget_request(url, repository_details)
 
         return unless body
 
         pages = body.fetch("items")
-        versions = Set.new
+        versions = T.let(Set.new, T::Set[String])
         pages.each do |page|
           items = page["items"]
           if items
             # inlined entries
-            items.each do |item|
```

Comment:
I split this up because rubocop was complaining about perceived complexity after I added an extra assertion to deal with a missing nil-check.